source/slang/slang-compiler.h

```diff
@@ -1922,6 +1922,34 @@ namespace Slang
     struct TypeCheckingCache;
     //
 
+    // Information about BaseType that's useful for checking literals 
+    struct BaseTypeInfo
+    {
+        typedef uint8_t Flags;
+        struct Flag 
+        {
+            enum Enum : Flags
+            {
+                Signed = 0x1,
+                FloatingPoint = 0x2,
+                Integer = 0x4,
+            };
+        };
+
+        SLANG_FORCE_INLINE static const BaseTypeInfo& getInfo(BaseType baseType) { return s_info[Index(baseType)]; }
+
+        static UnownedStringSlice asText(BaseType baseType);
+
+        uint8_t sizeInBytes;               ///< Size of type in bytes
+        Flags flags;
+        uint8_t baseType;
```

Comment:
We could in theory extend this to store the number of bits for various parts of the value, e.g.:

```
int signBits; // zero or one for unsigned/signed
int exponentBits; // zero for integers
int valueBits; // for floating point these are the mantissa bits
```

It would be possible in principle to derive the rest of what we need (minimum/maximum values, etc.) from those. Of course, that would rely on us having an arbitrary-precision numeric type to store the literal values...

Reply:
Well... there is a similar type in the stdlib. That just having such information defined in one place, would be preferable, but wasn't clear if that was appropriate with stdlib use.